edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/planners/KubernetesPlanner.cs

```diff
@@ -63,6 +63,12 @@ public class KubernetesPlanner : IPlanner
                 throw new InvalidModuleException($"Kubernetes deployment currently only handles type={typeof(DockerConfig).FullName}");
             }
 
+            // This is a workaround for K8s public preview refresh
+            // TODO: mark edgeagent module updateable when merging back to main release
+            if (desired.Modules.ContainsKey(Constants.EdgeAgentModuleName)) {
```

Comment:
I suggest we just exclude EdgeAgent module from the list of modules here instead of mark modules to be updatable.
```
// This is a workaround for K8s public preview refresh
// TODO: mark edgeagent module updateable when merging back to main release
desired = new ModulesSet(desired.Remove(Constants.EdgeAgentModuleName));
current = new ModulesSet(current.Remove(Constants.EdgeAgentModuleName));
```

It will prevent EdgeAgent to generate commands to Add/Update/Remove EdgeAgent module

Reply:
modules is an immutable dictionary in moduleset that we can't change.

Reply:
That's why I add the getter/setter in the module interface. And the getter/setter can be used for long term.

Reply:
ImmutableDictionary preserves its inside state right but when someone tries to modify it, operation returns a new instance of a dictionary that contains required change.
This is why I suggested creating a new instance of ModuleSets.
Here is some more info https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.iimmutabledictionary-2.remove?view=netcore-3.0#System_Collections_Immutable_IImmutableDictionary_2_Remove__0_

Reply:
I see. Thanks Dennis. 

If edgeagent status is changed, what's going to happen? Is it taking care of here?

Reply:
KubernetesRuntimeOperator watches pods inside namespaces and generates RuntimeInfo based on information about pods marked with a special label. Then it reports the status of the pod to iothub.
Other words, it is different path of EdgeAgent that should take care of status change for all modules (EdgeAgent included). We should be fine there.

But I agree, in the long run we may need this functionality in place. By that time we will have to come up with holistic approach though.

Reply:
Sounds good. Agree that it is risky making a core change for public preview refresh at this point.